sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuApproximatePercentile.scala

```diff
@@ -0,0 +1,220 @@
+  private lazy val (returnPercentileArray, percentiles) = makeArray(percentageExpression.value)
+
+  private def makeArray(v: Any): (Boolean, Either[Double, Array[Double]])  = v match {
+    // Rule ImplicitTypeCasts can cast other numeric types to double
+    case null => (false, Right(Array()))
+    case num: Double => (false, Left(num))
+    case arrayData: ArrayData => (true, Right(arrayData.toDoubleArray()))
+    case other => throw new IllegalStateException(s"Invalid percentile expression $other")
+  }
+
+  // The result type is the same as the input type.
+  private lazy val internalDataType: DataType = {
+    if (returnPercentileArray) {
+      ArrayType(child.dataType, containsNull = false)
+    } else {
+      child.dataType
+    }
+  }
+
+  override def dataType: DataType = internalDataType
+
+  override def children: Seq[Expression] = Seq(child, percentageExpression, accuracyExpression)
+}
+
+/**
+ * This expression computes an approximate percentile using a t-digest as input.
+ *
+ * @param child Expression that produces the t-digests.
+ * @param percentiles Percentile scalar, or percentiles array to evaluate.
+ * @param finalDataType Data type for results
+ */
+case class ApproxPercentileFromTDigestExpr(
+    child: Expression,
+    percentiles: Either[Double, Array[Double]],
+    finalDataType: DataType)
+  extends GpuExpression {
+
+  override def columnarEval(batch: ColumnarBatch): Any = {
+    val expr = child.asInstanceOf[GpuExpression]
+    withResource(expr.columnarEval(batch).asInstanceOf[GpuColumnVector]) { cv =>
+      percentiles match {
+        case Left(p) =>
+          // For the scalar case, we still pass cuDF an array of percentiles
+          // (containing a single item) and then extract the child column from the resulting
+          // array and return that (after converting from Double to finalDataType)
+          withResource(cv.getBase.approxPercentile(Array(p))) { percentiles =>
+            withResource(percentiles.extractListElement(0)) { childView =>
+              withResource(doCast(childView, DataTypes.DoubleType, finalDataType,
+                  ansiMode = SQLConf.get.ansiEnabled, legacyCastToString = false,
+                  stringToDateAnsiModeEnabled = SQLConf.get.ansiEnabled)) { childCv =>
+                GpuColumnVector.from(childCv.copyToColumnVector(), dataType)
+              }
+            }
+          }
+
+        case Right(p) =>
+          // array case - cast cuDF Array[Double] to Array[finalDataType]
+          withResource(cv.getBase.approxPercentile(p)) { percentiles =>
+            if (finalDataType == DataTypes.DoubleType) {
+              GpuColumnVector.from(percentiles.incRefCount(), dataType)
+            } else {
+              withResource(percentiles.getChildColumnView(0)) { childView =>
+                withResource(doCast(childView, DataTypes.DoubleType, finalDataType,
+                    ansiMode = SQLConf.get.ansiEnabled, legacyCastToString = false,
```

Comment:
Same here I don't think we should worry about ANSI mode, and instead just set it to false.